056-zk-kraft-migration.md

```diff
@@ -0,0 +1,435 @@
+The end of the migration will involve shutting down the ZooKeeper nodes reducing the resources consumption.
+
+Afterwards the migration, the users should be also able to migrate to use combined nodes for an additional reduction of resources, which is anyway out of the scope for this proposal.
+
+## Motivation
+
+The KRaft mode has been declared to be production-ready since Apache Kafka 3.3 release by the Apache Kafka project. 
+Anyway, there are still missing features as reported in the official [documentation](https://kafka.apache.org/documentation/#kraft_missing).
+The ZooKeeper mode has been deprecated after the Apache Kafka 3.5 version was released while providing the migration process to KRaft mode as preview.
+The coming Apache Kafka 3.6 release will have such a migration process supported as GA and the ZooKeeper mode will be supported until Apache Kafka 3.7 but removed after that.
+Starting from Apache Kafka 4.0 only KRaft mode will be supported.
+
+Based on the above Apache Kafka community plan, more details in [KIP-833](https://cwiki.apache.org/confluence/display/KAFKA/KIP-833%3A+Mark+KRaft+as+Production+Ready), the Strimzi project needs to be prepared to allow users to migrate their current clusters from running in ZooKeeper mode to KRaft mode.
+
+## Proposal
+
+### Prerequisites
+
+In order to start the ZooKeeper to KRaft migration, the following prerequisites should be met:
+
+* A Strimzi-managed Apache Kafka cluster running in ZooKeeper mode.
+* Running at Kafka 3.5.0 or higher and with `inter.broker.protocol.version` set to 3.5 or higher.
+* Configured to use one or more node pools, with `KafkaNodePool` custom resources, all with the `broker` role.
+* No existing node pool(s) which include the `controller` role.
+* If the Topic Operator is used, the `UnidirectionalTopicOperator` feature gate has to be enabled since the old Bidirectional Topic Operator does not work in the KRaft mode.
+* The `UseKRaft` feature gate should be enabled (explicitly, or via it being beta status).
+
+If the cluster is not configured to use node pools to run brokers, the user can follow the procedure described [here](https://strimzi.io/docs/operators/latest/deploying#proc-migrating-clusters-node-pools-str) in the official documentation, in order to meet such a prerequisite.
+
+
+### Migration
+
+The user controls the process by interacting with the `Kafka` and `KafkaNodePool` custom resources.
+The cluster operator (mostly via annotations and custom resources creation/deletion) performs the necessary operations: node re-configuration, node rolling, checking migration status, etc.
+The migration process can be rolled back to ZooKeeper at any point before taking the KRaft controllers out of the migration mode.
+
+### Finite State Machine (FSM)
+
+The migration process can be described with a Finite State Machine (FSM).
+The operator has a new `KRaftMigrationManager` component in charge of driving and tracking transitions in the FSM.
```

Comment:
This seems to be the only _implementation detail_. I would be a bit more curious about where and how it is used. But in any case, it seems a but weird to mention the name of some component without any further details. So I would either add more details (what does it do, where is it called, what classes use it and how etc.). Or simply not mention it at all. But in the current state it seems a bit weird as it drops what seems some class name without saying anything more.